### DISCUSSION TITLE
nuxt build --prerender vs routeRules Prerender

### DISCUSSION THREAD

**Discussion by chrispreisler** [nuxt/nuxt#29055] [Questions]
Hey guys,

I have a simple but for me not completely clear question. Is there any difference between

```
nuxt build --prerender
```

and

```
export default defineNuxtConfig({
  routeRules: {
    "/": { prerender: true },
  },
});
```

Thanks in advance for your help.

Best regards
Chris

**Comment by TheAlexLichter** [comment]
Actually, `nuxt build --prerender` and `nuxt generate` should be the same. See the following code:

https://github.com/nuxt/cli/blob/d5445ceca456c935c1653671887a08c3248dc5e8/src/commands/generate.ts#L19-L22
https://github.com/nuxt/cli/blob/d5445ceca456c935c1653671887a08c3248dc5e8/src/commands/build.ts#L32-L41

There is a differrence between `nuxt build` and `nuxt generate`/`nuxt build --prerender` though. If you use the former with your nuxt config, only / will be prerendered, the rest runs via SSR and a Node server. With `nuxt generate` you get only static files. I've [explained it also in this video](https://www.youtube.com/watch?v=SiT_1tfuPa4)

**Comment by chrispreisler** [comment]
Hey @manniL,

first of all, thank you very much for your answer and your video. But I still have a question.

I made a typo. I meant:

```
export default defineNuxtConfig({
  routeRules: {
    "/**": { prerender: true },
  },
});
```

So really all routes are included. And now my question is, when now using nuxt build, it doesn't build static files like with nuxt generate for the prerendered routes? So I mean it doesn't generate folders with the statics files and payload inside of them. At least I can't find them in the output build. Is this a technical limitation or do I understand something wrong? 

Thanks again for your help.

Best regards
Chris

**Comment by chrispreisler** [comment]
I made some more investigations and am now really not sure if this is a bug or not.

I have this as starter: https://stackblitz.com/edit/github-uycflm?file=nuxt.config.ts

Two pages Index and about, which are linked to each other.

When I build with this routeRules:

```
export default defineNuxtConfig({
  routeRules: {
    "/**": { prerender: true },
  },
});
```

There are no static routes generated.
When I change too:

```
export default defineNuxtConfig({
  routeRules: {
    "/": {
      prerender: true,
    },
    "/about": {
      prerender: true,
    },
  },
});
```

the static routes are generated. But as I understood with links I don't have to explicitily tell route rules which route rules as it crawls them for itself while building.

Is this a bug or do I miss any attribute in the nuxt.config.

Thanks again for your help.

Best regards
Chris
